src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1551,9 +1555,14 @@ ReturnCode_t DataWriterImpl::set_related_datareader(
     return ret;
 }
 
+// deadline_timer_reschedule returns true if it could compute and set a new interval; false if there’s no pending deadline.
 bool DataWriterImpl::deadline_timer_reschedule()
 {
     assert(qos_.deadline().period != dds::c_TimeInfinite);
+    if (deadline_timer_ == nullptr || deadline_missed_status_.total_count >= std::numeric_limits<uint32_t>::max())
+    {
+        return false;
+    }
```

Comment:
You don't need to check these conditions I believe, but should be done before calling the method. In any case, you could add the conditions to the assert.